integrations/datadog/client.py

```diff
@@ -233,6 +303,173 @@ async def get_slo_history(
+        """
+        Extracts environment names from the provided data structure.
+
+        Args:
+            data (dict): A dictionary containing tag information, potentially nested.
+
+        Returns:
+            list: A list of environment names (e.g., ['prod', 'staging']).
+        """
+        return [tag.split(":")[1] for tag in tags.keys() if tag.startswith("env:")]
+
+    async def _fetch_metrics_for_services(
+        self,
+        query: str,
+        envs_to_fetch: list[str],
+        services: list[dict[str, Any]],
+        time_window: int,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """Helper function to fetch metrics for a list of services and provided environments."""
+        logger.info(
+            f"Fetching metrics for {len(services)} services and {len(envs_to_fetch)} environments"
+        )
+
+        for service in services:
+            service_id = service["attributes"]["schema"]["dd-service"]
+
+            # Create tasks for concurrent fetching
+            tasks = []
+            for env_to_fetch in envs_to_fetch:
+                params = {"env": env_to_fetch, "service": service_id}
+                query_with_values = self._create_query_with_values(
+                    f"{query}{{service:{service_id},env:{env_to_fetch}}}", params
+                )
+
+                end_time = int(time.time())
+                start_time = end_time - (time_window * 60)
+
+                url = f"{self.api_url}/api/v1/query?from={start_time}&to={end_time}&query={query_with_values}"
+
+                # Create a task for each fetch operation
+                task = asyncio.create_task(
+                    self._fetch_with_rate_limit_handling(
+                        url, semaphore=self._metrics_semaphore
+                    )
+                )
+                tasks.append(task)
+
+            # Gather results concurrently
+            results = await asyncio.gather(*tasks)
+
+            # Process and yield results
+            metrics = []
+            for result, env_to_fetch in zip(results, envs_to_fetch):
+                # Update result with metadata
+                result.update(
+                    {
+                        SERVICE_KEY: service_id,
+                        METRIC_KEY: self.extract_metric_name(query),
+                        QUERY_ID_KEY: f"{query}/service:{service_id}/env:{env_to_fetch}",
+                        QUERY_KEY: query,
+                        ENV_KEY: env_to_fetch,
+                    }
+                )
+                metrics.append(result)
+
+            yield metrics
+
+    async def get_metrics(
+        self,
+        metric_query: str,
+        env: str = "*",
+        service: str = "*",
+        time_window: int = FETCH_WINDOW_TIME_IN_MINUTES,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """
+        Fetches metrics for specified services and environment.
+
+        Args:
+            metric (str): The Datadog metric to fetch.
```

Comment:
add example how `metric_query` looks like.
also align parameter naming in the docs

Reply:
alright